odoo/addons/base/transpiler/js_transpiler.py

```diff
@@ -0,0 +1,594 @@
+    repl = r"""\g<space>__exports[Symbol.for("default")] ="""
+    return EXPORT_DEFAULT_RE.sub(repl, new_content)
+
+
+IMPORT_BASIC_RE = re.compile(r"""
+    ^
+    (?P<space>\s*)                      # space and empty line
+    import\s+                           # import
+    (?P<object>{(\s*\w+\s*,?\s*)+})\s*  # { a, b, c as x, ... }
+    from\s*                             # from
+    (?P<path>(\".*\")|('.*')|(`.*`))    # "file path" ("some/path")
+    """, re.MULTILINE | re.VERBOSE)
+
+
+def convert_basic_import(content):
+    """
+    Transpile the simpler import call.
+
+    .. code-block:: javascript
+
+        // before
+        import { a, b, c as x } from "some/path"
+        // after
+        const {a, b, c: x} = require("some/path")
+    """
+    def repl(matchobj):
+        new_object = matchobj["object"].replace(" as ", ": ")
+        return f"{matchobj['space']}const {new_object} = require({matchobj['path']})"
+    return IMPORT_BASIC_RE.sub(repl, content)
+
+
+IMPORT_LEGACY_DEFAULT_RE = re.compile(r"""
+    ^
+    (?P<space>\s*)                                      # space and empty line
+    import\s+                                           # import
+    (?P<identifier>\w+)\s*                              # default variable name
+    from\s*                                             # from
+    (?P<path>(\"\w+\.\w+\")|('\w+\.\w+')|(`\w+\.\w+`))  # legacy alias file ("addon_name.module_name")
+    """, re.MULTILINE | re.VERBOSE)
+
+
+def convert_legacy_default_import(content):
+    """
+    Transpile legacy imports (that were used as they were default import).
+    Legacy imports means that their name is not a path but a <addon_name>.<module_name>.
+    It requires slightly different processing.
+
+    .. code-block:: javascript
+
+        // before
+        import module_name from "addon.module_name"
+        // after
+        const module_name = require("addon.module_name")
+    """
+    repl = r"""\g<space>const \g<identifier> = require(\g<path>)"""
+    return IMPORT_LEGACY_DEFAULT_RE.sub(repl, content)
+
+
+IMPORT_DEFAULT = re.compile(r"""
+    ^
+    (?P<space>\s*)                      # space and empty line
+    import\s+                           # import
+    (?P<identifier>\w+)\s*              # default variable name
+    from\s*                             # from
+    (?P<path>(\".*\")|('.*')|(`.*`))    # "file path" ("some/path")
+    """, re.MULTILINE | re.VERBOSE)
+
+
+def convert_default_import(content):
+    """
+    Transpile the default import call.
+
+    .. code-block:: javascript
+
+        // before
+        import something from "some/path"
+        // after
+        const something = require("some/path")[Symbol.for("default")]
+    """
+    repl = r"""\g<space>const \g<identifier> = require(\g<path>)[Symbol.for("default")]"""
+    return IMPORT_DEFAULT.sub(repl, content)
+
+
+RELATIVE_REQUIRE_RE = re.compile(r"""
+    require\((["'])([^@\"\']+)(["'])\) # require("some/path")
```

Comment:
```suggestion
    require\((["'])([^@"']+)(\1)\) # require("some/path")
```
This should ensure the quotes match properly. Also capturing the closing quote seems unnecessary: surely the start and end quote characters would be identical?

And I don't get why you're escaping the quotes in the middle group.